src/rules.jl

```diff
@@ -618,13 +642,14 @@ function check_rule(::Rule{:p2}, zxg::ZXGraph{T, P}, vs::Vector{T}) where {T, P}
     v1, v2 = vs
     if has_vertex(zxg.mg, v1)
         if spider_type(zxg, v1) == SpiderType.Z && is_interior(zxg, v1) &&
-            (degree(zxg, v1)) > 1 && (phase(zxg, v1) % 1//2 != 0) &&
-            (qubit_loc(zxg, v1) !== nothing || nqubits(zxg) == 0)
+            (degree(zxg, v1)) > 1 && (rem(phase(zxg, v1), 1//2) != 0) &&
+            length(neighbors(zxg, v1)) > 1
             if v2 in neighbors(zxg, v1)
                 if spider_type(zxg, v2) == SpiderType.Z && is_interior(zxg, v2) &&
-                    (phase(zxg, v2) == 0 || phase(zxg, v2) == 1) &&
-                    (qubit_loc(zxg, v2) !== nothing || nqubits(zxg) == 0)
-                    return true
+                    (phase(zxg, v2) in (0, 1))
+                    if all([length(neighbors(zxg, u)) > 1 for u in neighbors(zxg, v2)])
```

Comment:
no need to use `[]` just 
```suggestion
                    if all(length(neighbors(zxg, u)) > 1 for u in neighbors(zxg, v2))
```

Reply:
there is a few other places like this.